arbnode/resourcemanager/resource_management.go

```diff
@@ -31,37 +32,83 @@ var (
 // prior to RPC request handling.
 //
 // Must be run before the go-ethereum stack is set up (ethereum/go-ethereum/node.New).
-func Init(conf *Config) {
-	if conf.MemLimitPercent > 0 {
-		node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
-			var c limitChecker
-			c, err := newCgroupsMemoryLimitCheckerIfSupported(conf)
-			if errors.Is(err, errNotSupported) {
-				log.Error("No method for determining memory usage and limits was discovered, disabled memory limit RPC throttling")
-				c = &trivialLimitChecker{}
-			}
+func Init(conf *Config) error {
+	if conf.MemFreeLimit == "" {
+		return nil
+	}
+
+	limit, err := parseMemLimit(conf.MemFreeLimit)
+	if err != nil {
+		return err
+	}
 
-			return newHttpServer(srv, c), nil
+	node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
+		var c limitChecker
+		c, err := newCgroupsMemoryLimitCheckerIfSupported(limit)
+		if errors.Is(err, errNotSupported) {
+			log.Error("No method for determining memory usage and limits was discovered, disabled memory limit RPC throttling")
+			c = &trivialLimitChecker{}
 		}
+
+		return newHttpServer(srv, c), nil
 	}
+	return nil
+}
+
+func parseMemLimit(limitStr_ string) (int, error) {
+	limitStr := limitStr_
+	limitAccumulator := 1
+	done := false
+	for !done {
+		switch unicode.ToUpper(rune(limitStr[len(limitStr)-1])) {
+		case 'B':
+			limitStr = limitStr[:len(limitStr)-1]
+			continue
+		case 'K':
+			limitStr = limitStr[:len(limitStr)-1]
+			limitAccumulator *= 1024
+			done = true
+		case 'M':
+			limitStr = limitStr[:len(limitStr)-1]
+			limitAccumulator *= 1024 * 1024
+			done = true
+		case 'G':
+			limitStr = limitStr[:len(limitStr)-1]
+			limitAccumulator *= 1024 * 1024 * 1024
+			done = true
+		case 'T':
+			limitStr = limitStr[:len(limitStr)-1]
+			limitAccumulator *= 1024 * 1024 * 1024 * 1024
+			done = true
+		default:
+			done = true
+		}
+	}
+
+	limitInUnits, err := strconv.Atoi(limitStr)
+	if err != nil {
+		return 0, err
+	}
+
```

Comment:
It's not immediately clear that you use for loop to handle "B", "GB" case. You can do this a bit easier:

```lang=go
	var (
		limit int
		s string
	)
	fmt.Sscanf(limitStr, "%d%s", &limit, &s)
	switch (strings.ToUpper(s)) {
		case "K": 
			limit <<= 10   // Or just as you have it limit *= 1024, doesn't matter.
		case "M": 
			limit <<= 20
		case "G", "GB": 
			limit <<= 30
		case "T":
			limit <<= 40
	}
	if limit == 0 {
		// return some error here.
	}
	return limit
```

Reply:
Hah, thanks, my first cut at string parsing here was kind of ridiculous.